packages/python-google-compute-engine/google_compute_engine/networking/network_setup/network_setup.py

```diff
@@ -43,23 +43,45 @@ def __init__(self, dhclient_script=None, dhcp_command=None, debug=False):
     self.logger = logger.Logger(
         name='network-setup', debug=debug, facility=facility)
     self.distro_utils = distro_utils.Utils(debug=debug)
+    self.ipv6_initialized = False
 
   def EnableIpv6(self, interfaces):
     """Enable IPv6 on the list of network interfaces.
 
     Args:
       interfaces: list of string, the output device names for enabling IPv6.
     """
-    if not interfaces or set(interfaces) == self.ipv6_interfaces:
+    if not interfaces or self.ipv6_interfaces == set(interfaces):
       return
 
     self.logger.info('Enabling IPv6 on Ethernet interface: %s.', interfaces)
-    self.ipv6_interfaces = set(interfaces)
+    self.ipv6_interfaces = self.ipv6_interfaces.union(set(interfaces))
+    self.ipv6_initialized = True
 
     # Distro-specific setup for enabling IPv6 on network interfaces.
     self.distro_utils.EnableIpv6(
         interfaces, self.logger, dhclient_script=self.dhclient_script)
 
+  def DisableIpv6(self, interfaces):
```

Comment:
is this method ever called for anything other than the primary interface? i don't understand the instance attribute for ipv6_interfaces if it will only ever have one. we have at-change-time intent from EnableNetworkInterfaces, but this method also checks whether the interface is one of a set which have ipv6 configured. it's also not set in init so this method becomes unsafe if ever called before the enableipv6 method

Reply:
For this I am following the pattern you follow for the other interfaces, I move it to Init. But note you have other args outside of Init already.

Yes, for now this will only be called for the primary interface, but the code allows you to be future proof. The control of primary or not should not be done in this class.